test/functional/test_framework/segwit_addr.py

```diff
@@ -99,9 +99,20 @@ def decode(hrp, addr):
     return (data[0], decoded)
 
 
-def encode(hrp, witver, witprog):
+def encode_segwit_address(hrp, witver, witprog):
     """Encode a segwit address."""
     ret = bech32_encode(hrp, [witver] + convertbits(witprog, 8, 5))
-    if decode(hrp, ret) == (None, None):
+    if decode_segwit_address(hrp, ret) == (None, None):
         return None
     return ret
+
+class TestFrameworkScript(unittest.TestCase):
+    def test_segwit_encode_decode(self):
+        def test_python_bech32(addr):
+            hrp = addr[:4]
+            self.assertEqual(hrp, "bcrt")
+            (witver, witprog) = decode_segwit_address(hrp, addr)
+            self.assertEqual(encode_segwit_address(hrp, witver, witprog), addr)
+
+        test_python_bech32('bcrt1qqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqq3xueyj')
+        test_python_bech32('bcrt1qft5p2uhsdcdc3l2ua4ap5qqfg4pjaqlp250x7us7a8qqhrxrxfsqseac85')
```

Comment:
Previously this was testing p2wsh and p2wpkh, now it is only testing p2wsh

Reply:
Added a P2WPKH test